core/src/main/java/com/linecorp/armeria/server/ServiceRouteUtil.java

```diff
@@ -45,7 +46,6 @@ static RoutingContext newRoutingContext(ServerConfig serverConfig, Channel chann
                 routingStatus = RoutingStatus.INVALID_PATH;
             }
         } else {
-            pathAndQuery = PathAndQuery.parse(originalPath);
```

Comment:
The original if-else block is designed to minimize to call `PathAndQuery.parse(originalPath)`.
As `PathAndQuery` is given from the input parameter, should we simplify the code?
 
For example:
```java
if (pathAndQuery != null) {
    if (isCorsPreflightRequest(headers)) {
        routingStatus = RoutingStatus.CORS_PREFLIGHT;
    } else {
        routingStatus = RoutingStatus.OK;
    }
} else {
    if (headers.method() == HttpMethod.OPTIONS && "*".equals(originalPath)) {
        routingStatus = RoutingStatus.OPTIONS;
    } else {
        routingStatus = RoutingStatus.INVALID_PATH;
    }
}
```